WireAPI/Sources/WireAPI/APIs/TeamsAPI/TeamsAPIV0.swift

```diff
@@ -0,0 +1,367 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class TeamsAPIV0: TeamsAPI {
+
+    let httpClient: HTTPClient
+    let decoder = ResponsePayloadDecoder(decoder: .defaultDecoder)
+
+    init(httpClient: HTTPClient) {
+        self.httpClient = httpClient
+    }
+
+    var apiVersion: APIVersion {
+        .v0
+    }
+
+    func basePath(for teamID: Team.ID) -> String {
+        switch apiVersion {
+        case .v0:
+            "/teams/\(teamID.transportString())"
+        default:
+            "/v\(apiVersion.rawValue)/teams/\(teamID.transportString())"
+        }
+    }
+
+    // MARK: - Get team
+
+    func getTeam(for teamID: Team.ID) async throws -> Team {
+        let request = HTTPRequest(
+            path: basePath(for: teamID),
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        switch response.code {
```

Comment:
The response handling / error handling is repeated for most of these endpoints. Have you considered extracting it into a generic function?

Reply:
Yeah I'm looking into ways to reduce the duplication and boilerplate code.

Reply:
I factored out the boilerplate code into a response parser.